backend/curriculum_tracking/tests/test_api_views.py

```diff
@@ -156,6 +162,95 @@ def _test_set_due_time_permissions(self, card, get_progress):
         progress.refresh_from_db()
         self.assertEqual(progress.due_time.strftime("%c"), due_time_2.strftime("%c"))
 
+    def test_request_review_permissions(self):
+
+        # test using superuser
+        superuser = factories.UserFactory(is_superuser=True, is_staff=False)
+        content_item = factories.ProjectContentItemFactory(
+            project_submission_type=ContentItem.LINK, template_repo=None
+        )
+        card = factories.AgileCardFactory(
+            status=AgileCard.READY,
+            recruit_project=factories.RecruitProjectFactory(content_item=content_item),
+            content_item=content_item,
+        )
+        self.login(superuser)
+        card.start_project()
+        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
+        request_review_url = f"{self.get_instance_url(card.id)}request_review/"
+        response = self.client.post(request_review_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.IN_REVIEW)
+
+        # test using the card's assignee
+        content_item = factories.ProjectContentItemFactory(
+            project_submission_type=ContentItem.LINK, template_repo=None
+        )
+        card = factories.AgileCardFactory(
+            status=AgileCard.READY,
+            recruit_project=factories.RecruitProjectFactory(content_item=content_item),
+            content_item=content_item,
+        )
+        self.login(card.assignees.first())
+        card.start_project()
+        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
+        request_review_url = f"{self.get_instance_url(card.id)}request_review/"
+        response = self.client.post(request_review_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.IN_REVIEW)
+
+        # test using non superuser staff
+        staff_user = factories.UserFactory(is_superuser=False, is_staff=True)
+        content_item = factories.ProjectContentItemFactory(
+            project_submission_type=ContentItem.LINK, template_repo=None
+        )
+        card = factories.AgileCardFactory(
+            status=AgileCard.READY,
+            recruit_project=factories.RecruitProjectFactory(content_item=content_item),
+            content_item=content_item,
+        )
+        self.login(staff_user)
+        card.start_project()
+        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
+        request_review_url = f"{self.get_instance_url(card.id)}request_review/"
+        response = self.client.post(request_review_url)
+
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.data["detail"].code, "permission_denied")
+
+        card.refresh_from_db()
+        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
+
+        # test using not the cards assignee
```

Comment:
It looks like you are testing 4 different situations here. Should they all be in one function? Or should it be 4 functions? 
It also looks like ther eis a lot of repeated functionality. Maybe a lot of the setup could go in a setUp method?